c10/core/CPUAllocator.cpp

```diff
@@ -55,16 +55,19 @@ void* alloc_cpu(size_t nbytes) {
 #elif defined(_MSC_VER)
   data = _aligned_malloc(nbytes, gAlignment);
 #else
-  int err = posix_memalign(&data, gAlignment, nbytes);
-  if (err != 0) {
-    CAFFE_THROW(
-        "DefaultCPUAllocator: can't allocate memory: you tried to allocate ",
-        nbytes,
-        " bytes. Error code ",
-        err,
-        " (",
-        strerror(err),
-        ")");
+  // Allocate storage with malloc and manually align data pointer
+  // Then stash storage starting address before the first element to free later
+  // This is significantly faster than posix_memalign
```

Comment:
I imagine that this is dependent on the operating system and allocator in use; can you clarify where it was measured? In particular, I would expect jemalloc to do a decent job with posix_memalign...

Reply:
This was measured on ubuntu linux with default malloc.